### PR TITLE
Fetch api

### PR DESCRIPTION
**Pull request by elahe-mp**

### CODE REVIEW

File: src/Component/Pages/PostPage/PaginatedPosts.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const PaginatedPosts: React.FC = () => {
+  const [posts, setPosts] = useState<IPost[]>([]);
+  const [currentPageNo, setCurrentPageNo] = useState(1);
+  const postsPerPage = 10;
+  const [loadingData, setLoadingData] = useState(false);
+  const [error, setError] = useState<null | string>(null);
+
+  useEffect(() => {
+    setLoadingData(true);
+    fetch("https://jsonplaceholder.typicode.com/posts")
```
Comment (moh3n007) [inline src/Component/Pages/PostPage/PaginatedPosts.tsx:21]: Use paginated things of jsonplaceholder: 
https://jsonplaceholder.typicode.com/posts?_start=10&_limit=10

File: src/Component/Pages/PostPage/PaginatedPosts.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+const PaginatedPosts: React.FC = () => {
+  const [posts, setPosts] = useState<IPost[]>([]);
+  const [currentPageNo, setCurrentPageNo] = useState(1);
+  const postsPerPage = 10;
+  const [loadingData, setLoadingData] = useState(false);
+  const [error, setError] = useState<null | string>(null);
+
+  useEffect(() => {
+    setLoadingData(true);
```
Comment (moh3n007) [inline src/Component/Pages/PostPage/PaginatedPosts.tsx:20]: It's better to move these codes to a new function (like getPosts) with two props

File: src/Component/Pages/PostPage/PaginatedPosts.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+}
+
+const PaginatedPosts: React.FC = () => {
+  const [posts, setPosts] = useState<IPost[]>([]);
+  const [currentPageNo, setCurrentPageNo] = useState(1);
+  const postsPerPage = 10;
```
Comment (moh3n007) [inline src/Component/Pages/PostPage/PaginatedPosts.tsx:15]: It's better to create a constant out of the component

File: src/Component/Pages/PostPage/PaginatedPosts.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      });
+  }, []);
+
+  const handleUpdatePage = (inputValue: number) => {
+    setCurrentPageNo(inputValue);
+  };
+  //Get current posts
+  const indexOfLastPost = currentPageNo * postsPerPage;
+  const indexOfFirstPost = indexOfLastPost - postsPerPage;
+  const currentPosts = posts.slice(indexOfFirstPost, indexOfLastPost);
```
Comment (moh3n007) [inline src/Component/Pages/PostPage/PaginatedPosts.tsx:45]: It's better to use ```useMemo``` when you have a complicated variable. Don't forget to depend it to all variables you used inside of it

File: src/Component/Pages/TodoListPage/TodoList.tsx [hunk 5]
```diff
@@ -47,20 +74,31 @@ const TodoList: React.FC<ITodoList> = ({
...
+                    (
+                      inputValue: {
+                        todo: string;
+                        id: string;
+                        userName: string;
+                        completed: boolean;
+                      },
+                      index: number
+                    ) => (
                       <TableRow key={inputValue.id}>
```
Comment (moh3n007) [inline src/Component/Pages/TodoListPage/TodoList.tsx:95]: It's better to write a unique name, in this case : `TodoList_todo_${inputValue.id}`

Comment (elahe-mp) [reply src/Component/Pages/TodoListPage/TodoList.tsx]: refactor: modified the todoitems.map key as this: key={`TodoList_todo_${todo.id}`}

File: src/Component/Pages/PostPage/PaginatedPosts.tsx (hunk 4 above)
Comment (elahe-mp) [reply src/Component/Pages/PostPage/PaginatedPosts.tsx:45]: Used useMemo in PaginationRounded component.

File: src/Component/Pages/PostPage/PaginationRounded.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    () => Math.ceil(totalPosts / postsPerPage),
+    [totalPosts, postsPerPage]
+  );
+
+  const handleChange = (
+    e: React.ChangeEvent<unknown>,
+    currentPageNo: number
+  ) => {
+    handleUpdatePage(currentPageNo);
+    console.log("page no:", currentPageNo);
```
Comment (moh3n007) [inline src/Component/Pages/PostPage/PaginationRounded.tsx:29]: Please remove this line

**Comment by moh3n007** [comment]
Perfect!

**Pull request merged**